File: sixthSeminar/src/main/java/org/sopt/sixthSeminar/dto/response/post/PostGetResponse.java
package org.sopt.sixthSeminar.dto.response.post;

import org.sopt.sixthSeminar.domain.Category;
import org.sopt.sixthSeminar.domain.Post;
import org.sopt.sixthSeminar.service.CategoryService;

public record PostGetResponse(
        Long id,
        String title,
        String content,
        String category
) {
    public static PostGetResponse of(Post post, Category category) {
        return new PostGetResponse(post.getId(), post.getTitle(), post.getContent(), category.getContent());
    }
}

File: secondKotlin/src/main/kotlin/org/sopt/secondKotlin/dto/response/PostGetResponse.kt
package org.sopt.secondKotlin.dto.response

import org.sopt.secondKotlin.domain.Category
import org.sopt.secondKotlin.domain.Post

data class PostGetResponse(
    val id: Long,
    val title: String,
    val content: String,
    val category: String,
) {
    companion object {
        fun of(post: Post, category: Category): PostGetResponse {
            return PostGetResponse(
                post.id,
                post.title,
                post.content,
                category.content)
        }
    }
}